cmd/evm/internal/t8ntool/transition.go

```diff
@@ -292,6 +292,14 @@ func Transition(ctx *cli.Context) error {
 		prestate.Env.Difficulty = calcDifficulty(chainConfig, env.Number, env.Timestamp,
 			env.ParentTimestamp, env.ParentDifficulty, env.ParentUncleHash)
 	}
+	if chainConfig.IsCancun(big.NewInt(int64(prestate.Env.Number)), prestate.Env.Timestamp) {
+		// We require EIP-4788 beacon root to be set in the env
+		if env.ParentBeaconBlockRoot == nil {
+			return NewError(ErrorConfig, errors.New("post-cancun env requires parentBeaconBlockRoot to be set"))
+		}
+	} else {
+		env.ParentBeaconBlockRoot = nil // un-set it if it has been set too early
```

Comment:
Shouldn't it be an error if beacon root is requested before the fork?

Reply:
We _could_ but in go-ethereum, due to how the `json` parser works, we typically ignore extraneous json-fields. I think the engine-api is the sole exception where we explicitly check for, and error on, fields that we do not expect. 

For t8n, I'd prefer to keep the older behaviour